lib/tests/formulae_dependents.rb

```diff
@@ -9,20 +9,36 @@ def run!(args:)
         @source_tested_dependents = []
         @bottle_tested_dependents = []
 
-        (@testing_formulae - skipped_or_failed_formulae).each do |f|
-          dependent_formulae!(f, args: args)
+        @dependent_testing_formulae = @testing_formulae - skipped_or_failed_formulae
+        @dependent_testing_formulae.each do |formula_name|
+          install_formulae_if_needed_from_bottle!(formula_name)
+          puts
+        end
+
+        @dependent_testing_formulae.each do |formula_name|
+          dependent_formulae!(formula_name, args: args)
           puts
         end
       end
 
       private
 
+      def install_formulae_if_needed_from_bottle!(formula_name)
```

Comment:
I think this logic would be useful inside `tests/formulae.rb` too .

We can use actions/cache to cache built bottles and then skip the ones we already have them for. (Plus a flag to bust the cache if needed.)

Reply:
@carlocab `actions/cache` is way too small to be able to do this consistently so we may need another approach: https://github.blog/changelog/2021-11-23-github-actions-cache-size-is-now-increased-to-10gb-per-repository/

We can consider it in `tests/formulae.rb` when https://github.com/Homebrew/homebrew-core/pull/125595 and the new "bottles pushed to PRs" workflows are default and working nicely. It will be a much bigger lift.